### ISSUE TITLE
Failed tests (12884): TestSplitSnapshotRace_SnapshotWins

### ISSUE THREAD

**Issue by mjibson** [cockroachdb/cockroach#4415]
Appears unrelated to the change it was tested with.

https://circleci.com/gh/cockroachdb/cockroach/12884

```
I0216 20:36:59.080922 711 raft/raft.go:514  [group 11] 3 [logterm: 6, index: 28] sent vote request to 2 at term 42
I0216 20:36:59.081001 711 raft/raft.go:514  [group 11] 3 [logterm: 6, index: 28] sent vote request to 4 at term 42
I0216 20:36:59.081085 711 raft/raft.go:514  [group 11] 3 [logterm: 6, index: 28] sent vote request to 5 at term 42
W0216 20:36:59.081287 711 storage/replica.go:1204  group 2 on store 3 failed to send message to 4: storage/raft_transport.go:147: unknown peer 4
W0216 20:36:59.081402 711 storage/replica.go:1204  group 11 on store 3 failed to send message to 4: storage/raft_transport.go:147: unknown peer 4
--- FAIL: TestSplitSnapshotRace_SnapshotWins (13.29s)
	testing.go:168: storage/client_split_test.go:785: condition failed to evaluate within 10s: storage/client_test.go:708: expected [0 0 0 2 5 5], got [0 0 2 2 5 5]
	leaktest.go:149: Test appears to have leaked a store:
		github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc820089a40)
			/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:228 +0xd0e
		github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
			/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:212 +0x20
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8200e25b0, 0xc820015290)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:98 +0x52
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:99 +0x62
--
			/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x26
		main.main()
			github.com/cockroachdb/cockroach/storage/_test/_testmain.go:516 +0x113
I0216 20:36:59.086336 711 raft/raft.go:459  [group 11] 5 became follower at term 42
=== RUN   TestStoreSplitReadRace
F0216 20:36:59.086750 711 config/config.go:74  TestingDisableTableSplits called twice without cleaning up
goroutine 22891 [running]:
github.com/cockroachdb/cockroach/util/log.stacks(0xece557f01, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:877 +0xb8
github.com/cockroachdb/cockroach/util/log.(*loggingT).outputLogEntry(0x2562440, 0x3, 0x1f2161e, 0x10, 0x4a, 0xc821087a00, 0xc8207fd970)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:734 +0x12f
github.com/cockroachdb/cockroach/util/log.AddStructured(0x0, 0x0, 0x3, 0x2, 0x0, 0x0, 0xc821087a40, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/structured.go:38 +0x114
github.com/cockroachdb/cockroach/util/log.logDepth(0x0, 0x0, 0x1, 0x3, 0x0, 0x0, 0xc821087a40, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/log.go:64 +0x7b
github.com/cockroachdb/cockroach/util/log.Fatal(0xc821087a40, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/util/log/log.go:163 +0x5e
--
testing.tRunner(0xc82067a120, 0x2219aa8)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 1 [chan receive]:
testing.RunTests(0x1afcd68, 0x22185c0, 0xe4, 0xe4, 0x0)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc821cd9ef8, 0x1)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc821cd9ef8)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x21
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc821cd9ef8)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x26
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:516 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2562440)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 19860 [select]:
github.com/cockroachdb/cockroach/util/grpcutil.NewContextWithStopper.func1(0x7f7f5091b910, 0xc8214385c0, 0xc8200e25b0, 0xc8214266f0)
	/go/src/github.com/cockroachdb/cockroach/util/grpcutil/grpc_util.go:91 +0x107
created by github.com/cockroachdb/cockroach/util/grpcutil.NewContextWithStopper
	/go/src/github.com/cockroachdb/cockroach/util/grpcutil/grpc_util.go:96 +0x85

goroutine 19283 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc82077d9b0, 0xc8202ddf20, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/transport.go:141 +0x7e5
google.golang.org/grpc/transport.(*Stream).Read(0xc8203eeb60, 0xc8202ddf20, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/transport.go:294 +0x71
io.ReadAtLeast(0x7f7f5091ba98, 0xc8203eeb60, 0xc8202ddf20, 0x5, 0x5, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0x7f7f5091ba98, 0xc8203eeb60, 0xc8202ddf20, 0x5, 0x5, 0x7f7f5091bb98, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
google.golang.org/grpc.(*parser).recvMsg(0xc8202ddef0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/rpc_util.go:206 +0xc5
--
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc820b24e00, 0xc820c58180, 0x7f7f4c090880, 0xc820d0b7a0, 0xc8203eeb60)
	/go/src/google.golang.org/grpc/server.go:323 +0xa0
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/google.golang.org/grpc/server.go:324 +0x9a

goroutine 19668 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc82025baa0, 0xc82132a090, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/transport.go:141 +0x7e5
google.golang.org/grpc/transport.(*Stream).Read(0xc820ae3b20, 0xc82132a090, 0x5, 0x5, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/transport.go:294 +0x71
io.ReadAtLeast(0x7f7f5091ba98, 0xc820ae3b20, 0xc82132a090, 0x5, 0x5, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0x7f7f5091ba98, 0xc820ae3b20, 0xc82132a090, 0x5, 0x5, 0x7f7f5091bb98, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
google.golang.org/grpc.(*parser).recvMsg(0xc82132a060, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/rpc_util.go:206 +0xc5
--
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc820b24e00, 0xc820c58180, 0x7f7f4c090880, 0xc820d0b7a0, 0xc820ae3b20)
	/go/src/google.golang.org/grpc/server.go:323 +0xa0
created by google.golang.org/grpc.(*Server).serveStreams.func1
```

**Comment by mjibson** [comment]
Happened again on another PR: https://circleci.com/gh/cockroachdb/cockroach/12940

**Comment by tamird** [comment]
Fixed by #4427.
